From a11a67f81706bd1ed6f64464e110438d369b6376 Mon Sep 17 00:00:00 2001
From: Rico Ganahl <rico.ganahl@bytesatwork.ch>
Date: Wed, 12 Jul 2023 12:30:44 +0200
Subject: [PATCH] include: configs: bytedevkit enable oldi

Enable OLDI in u-boot to support LVDS display in Linux.
TI sysfw will fix this issue in a future release.

Signed-off-by: Rico Ganahl <rico.ganahl@bytesatwork.ch>
---
 include/configs/am62x_bytedevkit.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/include/configs/am62x_bytedevkit.h b/include/configs/am62x_bytedevkit.h
index fe77e4429a..1d60845262 100644
--- a/include/configs/am62x_bytedevkit.h
+++ b/include/configs/am62x_bytedevkit.h
@@ -92,6 +92,7 @@
 		"load mmc ${mmc_dev}:${mmc_part} ${loadaddr} ${bootdir}/${kernelfile} || exit; " \
 		"load mmc ${mmc_dev}:${mmc_part} ${dtbaddr} ${bootdir}/${dtbfile} || exit; " \
 		"run dtb_fixups; " \
+		"run enable_oldi; " \
 		"booti ${loadaddr} - ${dtbaddr}; " \
 	"\0" \
 	"update_emmc=echo Writing bootloader to eMMC; " \
@@ -103,6 +104,9 @@
 		"load mmc 1 ${loadaddr} u-boot.img; " \
 		"mmc write ${loadaddr} 0x1000 0x1000; " \
 		"mmc dev 0 0; " \
+	"\0" \
+	"enable_oldi=mw.l 0x00109008 0x68EF3490; mw.l 0x0010900C 0xD172BC5A; " \
+		"mw.l 0x00108700 0x00000000; " \
 	"\0"
 
 /* The remaining common defines, source <configs/ti_armv7_common.h> */
-- 
2.39.2

